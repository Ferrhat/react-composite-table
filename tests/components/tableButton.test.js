import React from 'react';
import { mount } from 'enzyme';
import TableButton from '../../src/components/tableButton';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('TableButton',()=>{
    const mockOnClick = jest.fn();
    const button = shallow(<TableButton className='icon icon-delete' onClick={mockOnClick} />);

    it('renders properly', () => {
        expect(button).toMatchSnapshot();
    });

    it('calls the onclick function', () => {
        expect(button.find('.list-inline-item > span').length).toEqual(1);
        button.find('.list-inline-item > span').simulate('click');

        expect(mockOnClick).toBeCalled();
    });
});
