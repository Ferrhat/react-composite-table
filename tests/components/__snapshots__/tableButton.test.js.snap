// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableButton renders properly 1`] = `
<li
  className="list-inline-item"
>
  <span
    className="icon icon-delete"
    onClick={[MockFunction]}
  />
</li>
`;
